packages/edit-post/src/components/layout/use-should-iframe.js

```diff
@@ -6,19 +6,13 @@ import { useSelect } from '@wordpress/data';
 import { store as blocksStore } from '@wordpress/blocks';
 import { store as blockEditorStore } from '@wordpress/block-editor';
 
-/**
- * Internal dependencies
- */
-import { store as editPostStore } from '../../store';
-
 const isGutenbergPlugin = globalThis.IS_GUTENBERG_PLUGIN ? true : false;
 
 export function useShouldIframe() {
```

Comment:
In which case the iframe is not used now? Can't we just remove this hook entirely

Reply:
Also WP.com users have been using iframe for all blocks (regardless of version) for block themes for a while. I think that represents a big number of users already, so I'm confident that there are very few blocks that actually break because of the iframe these days.

Reply:
If I recall correctly, when the post editor became an iframe in block themes (#46212), plugin developers reported issues (e.g. #47924). So the block API version 3 was introduced and the post editor was changed to not become an iframe, at least when v1/v2 blocks were registered (#48286).

And this PR doesn't just affect blocks: in fact, as mentioned later in [this comment](https://github.com/WordPress/gutenberg/pull/64351#issuecomment-2326301103), it seems there are still some well-known plugins that don't support the iframe editor.

Reply:
> And this PR doesn't just affect blocks: in fact, as mentioned later in https://github.com/WordPress/gutenberg/pull/64351#issuecomment-2326301103, it seems there are still some well-known plugins that don't support the iframe editor.

Do you know if these plugins actually don't support the iframe or just register blocks < 3. My suspicion is that most of the plugins that have blocks < v3 support the iframe but have trouble upgrading for other reasons.

Reply:
> Do you know if these plugins actually don't support the iframe or just register blocks < 3.

It seems that these plugins actually don't support the iframe. Please see the video at the end of [this comment](https://github.com/WordPress/gutenberg/pull/64351#issuecomment-2326301103).

Reply:
Is this about the "link" format?

Reply:
Yes.

Reply:
Ok, so it's not really about blocksV3, it seems like a separate discussion and these plugins seems like they're already broken for most new WP installs / site editor...

Reply:
That's right. From what I can see, these plugins don't enable their own link formatting in the post editor.

So the place these plugins don't actually work is in the mobile/tablet views which always run as an iframe editor. With this PR the desktop view will also run as an iframe editor, making the issue more apparent.